### PR TITLE
add profile grid view

### LINKED ISSUE #941: [FEATURE] Adding a grid view on profile
### Description

I want to add a grid view on profile page for feed like instagram. 

### Screenshots

Screenshot (623)

### Additional information

_No response_

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I have checked the existing issues
- [X] I'm a GSSoC'23 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by Palakkgoyal**
This PR closes #941 
I have added a grid view on profile page. When user click on any image they will be taken to the post view page of that particular post.

### CODE REVIEW

**Comment by narayan954** [comment]
Hi @Palakkgoyal I've been meaning to suggest 2 things to you regarding this project and open source
- while creating pr, you can use phrase like `solves #99999` or `fix #9999` or `fixes #7799989` (there are lot many like this, available on web) . using these phrases anywhere in comment of the pr links the pr automatically to that issue and hence closes the issue automatically on successfull merge. I'm editting this pr as well to show u an example. This is really convenient.
- There's a script `pretty` in the package.json in our project. Using that script you can automatically format all the files in our project(though there're some missing configuration that I will fix).Try using that script before committing to format everything.
To use that, run `npm run pretty`

File: src/pages/Profile/feed/index.css [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+@media screen and (min-width: 600px) {
+  .feed-main-container {
+    padding-top: 0;
+  }
+}
+
+@media screen and (max-width: 800px) {
+  .feed-main-container {
+    /* padding: 2em 10px 5.5em 10px; */
+  } /* @palak this is causing the issue for main feed */
```
Comment (narayan954) [inline src/pages/Profile/feed/index.css:76]: this is affecting the main feed apparently

**Review by narayan954 — COMMENTED** [review]
this is breaking the row mode(or column perhaps) of main/home feed, can you please have a look?

File: src/pages/Profile/feed/index.css (hunk 1 above)
Comment (narayan954) [reply src/pages/Profile/feed/index.css:76]: also I smell repetition of code XD

Comment (Palakkgoyal) [reply src/pages/Profile/feed/index.css]: Yes, I got the issue. The classes were same for profile feed and main feed. But, now I have done the changes. Can you please review them.

Comment (narayan954) [reply src/pages/Profile/feed/index.css]: sure, merging them! :)

**Review by narayan954 — APPROVED** [review]
looks good to me!

**Comment by narayan954** [comment]
i added some more changes, lmk how it looks now, cause I'm feeling happy after playing with css and able to not break Ui (perhaps) :')

**Comment by narayan954** [comment]
image
image

**Comment by Palakkgoyal** [comment]
Yes, It looks really better than before.

**Pull request merged**